### PR TITLE
Bug fix: L036 handle single-column `SELECT` with comment on same line as `SELECT` keyword

### LINKED ISSUE #3252: L036 - sqlfluff fix will remove a single SELECT target on a new line if there's a comment after the SELECT
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

If a violation of the rule **L036** (_Select targets should be on a new line unless there is only one select target_) is identified, with a single SELECT target on a new line but there's a double hyphen comment after the SELECT statement, `sqlfluff fix` will remove the new line resulting in the select target being appended to the end of the comment. 

### Expected Behaviour

For the following SQL query:

```SQL
SELECT --some comment
CASE WHEN a THEN 1 END;
```

I would expect the violation of L036 to be ignored, perhaps with a warning about the comment after the SELECT statement

```SQL
SELECT --some comment
  CASE WHEN a THEN 1 END;
```

### Observed Behaviour

However the following is observed:

```SQL
SELECT --some comment
CASE WHEN a THEN 1 END;
```

```SQL
SELECT --some comment CASE WHEN a THEN 1 END;
```

### How to reproduce

```bash
$ echo "SELECT --some comment\nCASE WHEN a THEN 1 END;" > test.sql
$ sqlfluff fix --dialect snowflake test.sql
```

### Dialect

Tested with:
**snowflake**
**sqlite**
**bigquery**

### Version

```bash
$ sqlfluff --version
sqlfluff, version 0.13.0
```

```bash
$ python --version
Python 3.9.6
```

### Configuration

Tested with default configurations but also with the following **.sqlfluff** config:

```
# https://docs.sqlfluff.com/en/stable/configuration.html#default-configuration
verbose = 0
nocolor = False
dialect = snowflake
templater = jinja
rules = None
exclude_rules = L011,L016,L031,L034,L035
recurse = 0
output_line_length = 120
runaway_limit = 10
ignore = None
ignore_templated_areas = True
encoding = autodetect
disable_noqa = False
sql_file_exts = .sql,.sql.j2,.dml,.ddl
fix_even_unparsable = False

[sqlfluff:indentation]
indented_joins = False
indented_ctes = False
indented_using_on = True
template_blocks_indent = True

[sqlfluff:templater]
unwrap_wrapped_queries = True

[sqlfluff:templater:jinja]
apply_dbt_builtins = True

[sqlfluff:templater:jinja:macros]
# Macros provided as builtins for dbt projects
dbt_ref = {% macro ref(model_ref) %}{{model_ref}}{% endmacro %}
dbt_source = {% macro source(source_name, table) %}{{source_name}}_{{table}}{% endmacro %}
dbt_config = {% macro config() %}{% for k in kwargs %}{% endfor %}{% endmacro %}
dbt_var = {% macro var(variable, default='') %}item{% endmacro %}
dbt_is_incremental = {% macro is_incremental() %}True{% endmacro %}

# Some rules can be configured directly from the config common to other rules.
[sqlfluff:rules]
tab_space_size = 2
max_line_length = 120
indent_unit = space
comma_style = leading
allow_scalar = True
single_table_references = consistent
unquoted_identifiers_policy = all

# Some rules have their own specific config.
# https://docs.sqlfluff.com/en/stable/rules.html
[sqlfluff:rules:L003]
hanging_indents = False

[sqlfluff:rules:L007]  # Keywords
operator_new_lines = after

[sqlfluff:rules:L010]  # Keywords
capitalisation_policy = upper

[sqlfluff:rules:L011]  # Table aliasing
aliasing = implicit

[sqlfluff:rules:L012]  # Column aliasing
aliasing = explicit

[sqlfluff:rules:L013]  # Column expressions
allow_scalar = False

[sqlfluff:rules:L014]  # Unquoted identifiers
extended_capitalisation_policy = lower

[sqlfluff:rules:L016]
ignore_comment_lines = False

[sqlfluff:rules:L029]  # Keyword identifiers
unquoted_identifiers_policy = aliases

[sqlfluff:rules:L030]  # Function names
extended_capitalisation_policy = upper

[sqlfluff:rules:L038]
select_clause_trailing_comma = forbid

[sqlfluff:rules:L040]  # Null & Boolean Literals
capitalisation_policy = consistent

[sqlfluff:rules:L042]
# By default, allow subqueries in FROM clauses, but not JOIN clauses.
forbid_subquery_in = join

[sqlfluff:rules:L047]  # Consistent syntax to count all rows
prefer_count_1 = False
prefer_count_0 = False

[sqlfluff:rules:L052]  # Statements must end with a semi-colon
multiline_newline = True
require_final_semicolon = True

[sqlfluff:rules:L059] # Unnecessary quoted identifier
ignore_words = date%
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by barrywhart**
### Brief summary of the change made
Fixes #3252

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/rules/L036.py [hunk 1]
```diff
@@ -301,12 +315,7 @@ def _fixes_for_move_after_select_clause(
                         ]
                         fixes_.append(
                             LintFix.create_after(
-                                self._choose_anchor_segment(
-                                    context,
-                                    "create_after",
-                                    select_clause[0],
-                                    filter_meta=True,
-                                ),
+                                select_clause[0],
```
Comment (barrywhart) [inline src/sqlfluff/rules/L036.py:318]: Unrelated to the issue, but no longer necessary to call `_choose_anchor_segment()` here due to recent improvements in the core linter.

**Comment by tunetheweb** [comment]
Instead of moving the select target _up_, before the comment, could we move the comment _down_ after the select target (or temporary delete it)? Then fix the select target as it would if not comment was there?

File: src/sqlfluff/rules/L036.py [hunk 2]
```diff
@@ -384,15 +392,25 @@ def _fixes_for_move_after_select_clause(
                             ],
                         )
 
-            fixes += [
```
Comment (barrywhart) [inline src/sqlfluff/rules/L036.py:387]: Review this section with "Hide whitespace" enabled.

**Comment by barrywhart** [comment]
@tunetheweb: We could. I don't think I'm up for tackling it now. Wna tocreate a separate issue for this? Often when I look at one of these rules that moves stuff around, I am surprised how complicated it is and how many special cases arise, and how easy it is to make a mistake and corrupt the code. Here are a couple of other test cases that occur to me if we try to handle this:
```
SELECT -- comment 1
a -- comment 2;

SELECT --comment 1
/* comment 2 */ a -- comment 3;
```

🤯

**Comment by tunetheweb** [comment]
Fair enough!

**Review by tunetheweb — APPROVED** [review]

**Pull request merged**